### PR TITLE
Use time limits instead of number iterations limits in GHA performance tests

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
This PR should improve the pre-commit stability. 

 - use performance counter instead of time() for measurement
 - heat and measure based on time limit instead of max num iterations
 - fix html output results

### Tickets:
 - 126474

### CODE REVIEW

File: tests/model_hub_tests/performance_tests/conftest.py [hunk 1]
```diff
@@ -26,27 +27,32 @@ def pytest_runtest_makereport(item, call):
...
-
-def round_num(n: float) -> str:
-    s = '{:.4E}'.format(n)
-    if s.endswith('E+00'):
-        return s[:-4]
-    return s
+    cells.insert(5, html.th('converted model heat n repeats'))
+    cells.insert(6, html.th('converted model measurement n repeats'))
+    cells.insert(7, html.th('converted model throughput'))
+    cells.insert(8, html.th('read model infer time secs'))
```
Comment (itikhono) [inline tests/model_hub_tests/performance_tests/conftest.py:35]: I believe we should leave only "infer time", "variance" and maybe "measurement repeats" in the html report

**Review by itikhono — APPROVED** [review]

File: tests/model_hub_tests/models_hub_common/constants.py [hunk 2]
```diff
@@ -5,6 +5,10 @@
 import os
 import tempfile
 
+runtime_measure_duration = os.environ.get('RUNTIME_MEASURE_DURATION', '60')
+runtime_heat_duration = os.environ.get('RUNTIME_HEAT_DURATION', '5')
```
Comment (rkazants) [inline tests/model_hub_tests/models_hub_common/constants.py:9]: can you add a comment? Is it in minutes or seconds?

File: tests/model_hub_tests/models_hub_common/test_performance_model.py [hunk 3]
```diff
@@ -124,22 +134,41 @@ def get_read_model(self, model_path: str):
...
+        left_time = runtime_duration_ns
+        time_slices = []
+        while left_time > 0:
+            t0 = time.perf_counter_ns()
+            ov_model(inputs)
+            t1 = time.perf_counter_ns()
+            timedelta = t1 - t0
+            time_slices.append(timedelta)
+            left_time -= timedelta
+            results.infer_n_repeats += 1
```
Comment (rkazants) [inline tests/model_hub_tests/models_hub_common/test_performance_model.py:159]: please put this performance measurement routine into separate function or wrapper. We will use it for other purpose to measure original FW performance

File: tests/model_hub_tests/models_hub_common/test_performance_model.py [hunk 4]
```diff
@@ -124,22 +134,41 @@ def get_read_model(self, model_path: str):
...
     def infer_model(self, ov_model, inputs):
+        results = ModelResults()
         infer_step_t0 = time.time()
         # heat run
-        for _ in range(0, TestModelPerformance.num_heat_runs):
-            ov_model(inputs)
-        # measure
-        results = []
-        for _ in range(0, TestModelPerformance.num_measure_runs):
+        left_time = float(runtime_heat_duration)
```
Comment (rkazants) [inline tests/model_hub_tests/models_hub_common/test_performance_model.py:140]: please make a comment that you heat hardware before actual measurement

File: tests/model_hub_tests/models_hub_common/test_performance_model.py [hunk 5]
```diff
@@ -124,22 +134,41 @@ def get_read_model(self, model_path: str):
...
+        results.infer_mean_time = np.mean(time_slices)
+        results.infer_variance = (np.std(time_slices, ddof=1) * 100) / results.infer_mean_time
         infer_step_t1 = time.time()
-        print('inference measurement done in {} secs'.format(infer_step_t1 - infer_step_t0))
-        return mean, var
+        print('model time infer {}'.format(round_num(results.infer_mean_time / (10 ** 9))))
+        print('model time infer var {}'.format(round_num(results.infer_variance)))
+        print('model time infer throughput {}'.format(round_num(results.infer_throughput)))
+        print('inference measurement done in {} repeats'.format(results.infer_n_repeats))
+        print('inference measurement done in {} secs'.format(round_num(infer_step_t1 - infer_step_t0)))
```
Comment (rkazants) [inline tests/model_hub_tests/models_hub_common/test_performance_model.py:170]:
```suggestion
        print('Model inference average time: {} sec.'.format(round_num(results.infer_mean_time / (10 ** 9))))
        print('Model inference time variance: {} %'.format(round_num(results.infer_variance)))
        print('Model inference throughput: {} times/sec.'.format(round_num(results.infer_throughput)))
```

Comment (rkazants) [reply tests/model_hub_tests/models_hub_common/test_performance_model.py:170]: Let us simplify output and clarify (make if clear). Not sure that all lines are needed

File: tests/model_hub_tests/models_hub_common/test_performance_model.py [hunk 6]
```diff
@@ -124,22 +134,41 @@ def get_read_model(self, model_path: str):
...
+            t0 = time.perf_counter_ns()
+            ov_model(inputs)
+            t1 = time.perf_counter_ns()
+            timedelta = t1 - t0
+            time_slices.append(timedelta)
+            left_time -= timedelta
+            results.infer_n_repeats += 1
+        print('measurement done in {} repeats'.format(results.infer_n_repeats))
+        runtime_real = runtime_duration_ns - left_time
+        results.infer_throughput = runtime_real / (results.infer_n_repeats * (10 ** 9))
```
Comment (rkazants) [inline tests/model_hub_tests/models_hub_common/test_performance_model.py:162]: Is it sec./times? I though it should be in times/sec.

File: tests/model_hub_tests/models_hub_common/utils.py [hunk 7]
```diff
@@ -63,3 +65,10 @@ def cleanup_dir(dir: str):
                 shutil.rmtree(file_path)
         except Exception as e:
             pass
+
+
+def round_num(n: float) -> str:
+    s = '{:.4E}'.format(n)
+    if s.endswith('E+00'):
+        return s[:-4]
```
Comment (rkazants) [inline tests/model_hub_tests/models_hub_common/utils.py:73]: let us leave just two digits after dot. Now it looks long

**Comment by rkazants** [comment]
Also, please work on the report to make it better.

Now it looks as follows:
image

Several comments how to improve:
1) Status column value repeats a name of the column: `Status: OK`. Let us just leave: `OK`. Now it looks not cute:). And let us call column name: `status` -> `Status`
2) `converted model infer time secs` -> `convert_model Pipeline Inference Time, sec.`
3) converted model infer time variance - NOT needed, it can be found in logs if someone wants to deep dive. please remove
4) `converted model measurement n repeats` - NOT needed, it can be found in logs if someone wants to deep dive. please remove
5) `read model infer time secs` -> `read_model Pipeline Inference Time, sec.`
6) read model infer time variance - NOT needed, it can be found in logs if someone wants to deep dive. please remove
7) read model measurement n repeats - NOT needed, it can be found in logs if someone wants to deep dive. please remove
8) `model infer time ratio converted_model_time/read_model_time` -> `Inference Time Ratio (convert_model vs. read_model)`
9) Links - NOT NEEDED. please remove

**Review by rkazants — APPROVED** [review]

**Pull request merged**